File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsWallet.swift
import Foundation

// MARK: - Credentials wallet (M12.2)

struct CCRAchievement: Codable, Identifiable, Hashable {
    var id: String
    var type: String
    var title: String
    var description: String?
    var issuedAt: String
    var evidenceUrl: String?
    var outcomeTags: [String]?
}

struct CCRDocument: Codable, Identifiable, Hashable {
    var id: String
    var generatedAt: String
    var shareable: Bool
    var verificationUrl: String?
}

struct CCRSummaryResponse: Codable {
    var achievements: [CCRAchievement]?
    var documents: [CCRDocument]?
}

struct CCRGenerateRequest: Encodable {
    var sharePublicly: Bool
}

struct CCRGenerateResponse: Decodable {
    var document: CCRDocument
    var achievements: [CCRAchievement]?
    var verificationUrl: String?
}

struct CETranscriptAward: Codable, Identifiable, Hashable {
    var courseTitle: String
    var ceuCredit: Double
    var contactHours: Double
    var completedAt: String

    var id: String { "\(courseTitle)-\(completedAt)" }
}

struct CETranscriptResponse: Codable {
    var awards: [CETranscriptAward]?
}

struct TranscriptRequestSummary: Codable, Identifiable, Hashable {
    var id: String
    var status: String
    var deliveryType: String
    var deliveryEmail: String?
    var deliveryAddress: String?
    var urgencyDays: Int?
    var urgencyDaysMin: Int?
    var urgencyUnit: String?
    var requestedAt: String
    var submittedAt: String?
    var errorMessage: String?
    var webhookResponseCode: Int?
}

struct TranscriptRequestsResponse: Codable {
    var requests: [TranscriptRequestSummary]?
}

struct TranscriptsStudentConfig: Codable {
    var pickupInstructions: String?
    var pickupAvailable: Bool
}

struct TranscriptRequestCreateBody: Encodable {
    var deliveryType: String
    var deliveryEmail: String?
    var deliveryAddress: String?
    var mailUrgency: String?
    var urgencyDays: Int?
}

struct TranscriptRequestCreateResponse: Decodable {
    var request: TranscriptRequestSummary?
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsWallet.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class CCRAchievement(
    val id: String,
    val type: String,
    val title: String,
    val description: String? = null,
    val issuedAt: String,
    val evidenceUrl: String? = null,
    val outcomeTags: List<String>? = null,
)

@Serializable
data class CCRDocument(
    val id: String,
    val generatedAt: String,
    val shareable: Boolean = false,
    val verificationUrl: String? = null,
)

@Serializable
data class CCRSummaryResponse(
    val achievements: List<CCRAchievement>? = null,
    val documents: List<CCRDocument>? = null,
)

@Serializable
data class CCRGenerateRequest(
    val sharePublicly: Boolean,
)

@Serializable
data class CCRGenerateResponse(
    val document: CCRDocument,
    val achievements: List<CCRAchievement>? = null,
    val verificationUrl: String? = null,
)

@Serializable
data class CETranscriptAward(
    val courseTitle: String,
    val ceuCredit: Double,
    val contactHours: Double,
    val completedAt: String,
)

@Serializable
data class CETranscriptResponse(
    val awards: List<CETranscriptAward>? = null,
)

@Serializable
data class TranscriptRequestSummary(
    val id: String,
    val status: String,
    val deliveryType: String,
    val deliveryEmail: String? = null,
    val deliveryAddress: String? = null,
    val urgencyDays: Int? = null,
    val urgencyDaysMin: Int? = null,
    val urgencyUnit: String? = null,
    val requestedAt: String,
    val submittedAt: String? = null,
    val errorMessage: String? = null,
    val webhookResponseCode: Int? = null,
)

@Serializable
data class TranscriptRequestsResponse(
    val requests: List<TranscriptRequestSummary>? = null,
)

@Serializable
data class TranscriptsStudentConfig(
    val pickupInstructions: String? = null,
    val pickupAvailable: Boolean = false,
)
